src/extension.ts

```diff
@@ -57,11 +57,14 @@ export async function activateInternal(context: vscode.ExtensionContext, perfSta
         context.subscriptions.push(ext.appResourceTree.trackTreeItemCollapsibleState(ext.appResourceTreeView));
 
         // Hook up the resolve handler
-        registerEvent('treeItem.expanded', ext.appResourceTree.onDidExpandOrRefreshExpandedTreeItem, async (context: IActionContext, treeItem: AzExtTreeItem) => {
+        registerEvent('treeItem.expanded', ext.appResourceTree.onDidExpandOrRefreshExpandedTreeItem, async (context: IActionContext, data: OnDidExpandOrRefreshExpandedEmitterData) => {
             context.errorHandling.suppressDisplay = true;
-
-            if (treeItem instanceof GroupTreeItemBase) {
-                await treeItem.resolveAllChildrenOnExpanded(context);
+            if (data.item instanceof GroupTreeItemBase) {
+                if (data.source !== 'refresh') {
+                    // clear VS Code's internal cache
+                    ext.appResourceTree.refreshUIOnly(undefined);
```

Comment:
I don't think I understand why we should clear VSCode's cache when expanding a node.

Reply:
Focusing, then unfocusing a group puts the tree view in a state in which expanding group nodes causes the "Tree item with id X is already registered" error. 

My theory is: on focus the app resource tree items are not removed from VS Code's tree item cache, but the group tree items are removed. So VS Code's cache contains the children, but not the parents. On unfocus, we give VS Code the parents (group tree items). 

Then, when a group tree item is expanded VS Code doesn't associate the app resources that were never removed from the cache on focus with the new group tree items, and throws an error when `getChildren` returns items which already exist in VS Code's cache.

The best fix is to make sure the app resources are cleared from the cache on focus/unfocus. I tried a bunch of things but was never able to get this to happen. `refreshUIOnly` is already [called upon every focus/unfocus command](https://github.com/microsoft/vscode-azureresourcegroups/blob/a1290599aa1e7ea771de36c0d830e7b33cdd6cc0/src/tree/SubscriptionTreeItem.ts#L217). To me this should completely clear VS Code's cache on every focus/unfocus. However it doesn't seem to be doing that.

So I ended up on this "band aid" solution.